app/controllers/brapi/v1/germplasm_controller.rb

```diff
@@ -0,0 +1,250 @@
+  end
+
+
+
+
+ def search
+    # accepted params: germplasmPUI , germplasmDbId , germplasmName , pageSize , page
+    # domainParameters = ['germplasmPUI','germplasmDbId','germplasmName']
+    # totalParameters = domainParameters + ['pageSize','page']
+    
+    if !params['germplasmPUI'].present? && !params['germplasmDbId'].present? && !params['germplasmName'].present?
+      render json: { reason: 'Resource not found' }, status: :not_found
+    else 
+      result_object = germplasm_query(params, false)
+      records = result_object.values
+      
+      
+      if records.nil?
+        render json: { reason: 'Resource not found' }, status: :not_found
+      else
+        if records.size == 1 && records[0]!=nil && records[0] != api_key.user_id && records[1] 
+          render json: { reason: 'This is a private resource of another user' }, status: :unauthorized
+        else
+          json_result_array = []
+          
+          # any programmatic data manipulation can be done here
+          result_object.each do |row|
+            row = JSON.parse(row["row_to_json"])
+            puts row
+            if (!row[:user_id] || (row[:user_id]!=nil && row[:user_id] != api_key.user_id && row[:published] ))
+              json_result_array << row
+            end
+          end
+          
+          # pagination data returned
+          page = get_page
+          pageSize = get_pageSize 
+          
+          result_count_object = germplasm_query(params, true)
+          totalCount = result_count_object.values.first[0].to_i
+          totalPages = (totalCount/pageSize.to_f).ceil
+          
+          json_response = { 
+            'metadata' => json_metadata(pageSize, page, totalCount, totalPages),
+            'result' => json_result_array
+          }
+         
+          render json: json_response, :except => ["id", "user_id", "created_at", "updated_at","total_entries_count"]
+        end
+      end
+      
+    
+    end
+  end
+
+
+
+
+  private
+
+
+
+  # processing result: https://github.com/plantbreeding/API/blob/master/Specification/Germplasm/GermplasmSearchGET.md 
+  # All possible fields to return:
+  # germplasmDbId, defaultDisplayName, accessionNumber, germplasmName, germplasmPUI, pedigree, 
+  # seedSource, synonyms, commonCropName, instituteCode, instituteName, biologicalStatusOfAccessionCode, 
+  # countryOfOriginCode, typeOfGermplasmStorageCode, genus, species, speciesAuthority, subtaxa, subtaxaAuthority, 
+  # donors, acquisitionDate
+
+  def germplasm_query(params, count_mode)
+    # where conditions
+    where_query = " "
+    where_atts = []
+    where_atts_count = 0
+    
+    # TO BE DEFINED
+    #if params['germplasmPUI'].present? 
+    
+    if params['germplasmDbId'].present?   # plant_accessions.plant_accession
+      where_atts_count+= 1
+      where_query = where_query + "where plant_accessions.plant_accession = $"+where_atts_count.to_s
+      where_atts<< params['germplasmDbId']
```

Comment:
If I am correct, you are putting user-provided parameters "as are" inside the query string. If this is the case, please add some code to check the `params` content against code injection.